### PR TITLE
Optimize android instrumentation test time

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
Using auto shard strategy from this doc https://www.browserstack.com/docs/app-automate/espresso/test-sharding
Since we've a limit of 5 parallel tests, so I've configured the `numberOfShards` to 5.

### CODE REVIEW

**Review by shubham1g5 — DISMISSED** [review]
awesome

File: app/instrumentation-tests/src/org/commcare/androidTests/MenuTests.kt [hunk 1]
```diff
@@ -24,7 +24,6 @@ import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 @LargeTest
-@BrowserstackTests
```
Comment (shubham1g5) [inline app/instrumentation-tests/src/org/commcare/androidTests/MenuTests.kt:27]: annotation on class doesn't work ?

**Review by shubham1g5 — APPROVED** [review]

File: app/instrumentation-tests/src/org/commcare/androidTests/MenuTests.kt (hunk 1 above)
Comment (ShivamPokhriyal) [reply app/instrumentation-tests/src/org/commcare/androidTests/MenuTests.kt:27]: It works. `BrowserstackTests` means that the tests will run on browserstack. and we don't want to run all the tests from this class on browserstack. So I've annotated the tests which need to run there.

Comment (shubham1g5) [reply app/instrumentation-tests/src/org/commcare/androidTests/MenuTests.kt:27]: Ahh I completely misunderstood the intent here. Thanks for clarification.

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
I guess we have some tests `CaseSharingTest` which needs to run in order. And with shard enabled, these tests will run on different devices and would fail. 
So perhaps we can use one shard to run these sorts of tests(maybe by using annotation or specifying the class directly).

**Comment by shubham1g5** [comment]
I think 2 diiferent tests should be independent of each other and if they are dependent they should be merged into a single test.

**Comment by ShivamPokhriyal** [comment]
> I think 2 diiferent tests should be independent of each other and if they are dependent they should be merged into a single test.

Yeah, I wasn't really sure about that and they're 5 methods that need to be run in specific order. So, I'll merge them into a big one then.

**Comment by dimagimon** [comment]
[32% (0.0%) vs master 32%](https://jenkins.dimagi.com/job/commcare-android-pr-job-builder/462147/)

**Review by shubham1g5 — APPROVED** [review]

**Comment by shubham1g5** [comment]
Nice this last build was just ~25 mins.

**Pull request merged**